lib/bamboo/adapters/sendgrid_adapter.ex

```diff
@@ -147,6 +148,17 @@ defmodule Bamboo.SendgridAdapter do
   defp put_text_body(body, %Email{text_body: nil}), do: body
   defp put_text_body(body, %Email{text_body: text_body}), do: Map.put(body, :text, text_body)
 
+  defp put_x_smtp_api(body, %Email{private: %{"x-smtpapi" => fields}} = email) do
+    # SendGrid will error with empty bodies, even while using templates.
+    # Sets a default `text_body` if one is not specified, allowing the consumer
+    # to negelect doing so themselves.
```

Comment:
Usually I prefer no comments, but in this case I think this is a good idea!

There is one typo though. negelect -> neglect